src/states/StitchedContentPlayback.js

```diff
@@ -0,0 +1,44 @@
+import {ContentState, StitchedAdRoll} from '../states.js';
+
+/*
+ * This state represents content playback when stitched ads are in play.
+ */
+export default class StitchedContentPlayback extends ContentState {
+
+  /*
+   * Allows state name to be logged even after minification.
+   */
+  static _getName() {
+    return 'StitchedContentPlayback';
+  }
+
+  /*
+   * For state transitions to work correctly, initialization should
+   * happen here, not in a constructor.
+   */
+  init() {
+
+    // Don't block calls to play in stitched ad players, ever.
+    this.player.ads._shouldBlockPlay = false;
+
+    // Tell redispatch not to intercept ended events. This should prevent
+    // the firing of readyforpostroll.
+    this.player.ads._contentHasEnded = true;
```

Comment:
Sounds good to me.

Reply:
Seems like we can add a check in redispatch for when we see `ended` with stitched ads, but we do still want to get an `adended` event so the states can handle that and get us out of ad mode before re-triggering `ended`. Updating soon.

Reply:
If we return early in redispatch, it will allow the original ended event to not be cancelled or redispatched and you won't need to retrigger it at all.